src/webgpu/shader/execution/statement/increment_decrement.spec.ts

```diff
@@ -0,0 +1,301 @@
+      `
+    var a: i32 = -10;
+    var b: i32 = 10;
+    var c: i32 = ${kValue.i32.negative.min};
+    var d: i32 = ${kValue.i32.positive.max - 1};
+    var e: i32 = 0;
+
+    a++;
+    b++;
+    c++;
+    d++;
+    e++;
+
+    push_output(a);
+    push_output(b);
+    push_output(c);
+    push_output(d);
+    push_output(e);
+`
+    );
+  });
+
+g.test('scalar_i32_increment_overflow')
+  .desc('Tests increment of scalar i32 values which overflows')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'i32',
+      new Int32Array([kValue.i32.negative.min]),
+      `
+    var a: i32 = ${kValue.i32.positive.max};
+    a++;
+    push_output(a);
+`
+    );
+  });
+
+g.test('scalar_u32_increment')
+  .desc('Tests increment of scalar u32 values')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'u32',
+      new Uint32Array([1, 11, kValue.u32.max]),
+      `
+    var a: u32 = 0;
+    var b: u32 = 10;
+    var c: u32 = ${kValue.u32.max - 1};
+
+    a++;
+    b++;
+    c++;
+
+    push_output(a);
+    push_output(b);
+    push_output(c);
+`
+    );
+  });
+
+g.test('scalar_u32_increment_overflow')
+  .desc('Tests increment of scalar u32 values which overflows')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'u32',
+      new Uint32Array([0]),
+      `
+    var a: u32 = ${kValue.u32.max};
+    a++;
+    push_output(a);
+`
+    );
+  });
+
+g.test('scalar_i32_decrement')
+  .desc('Tests decrement of scalar i32 values')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'i32',
+      new Int32Array([-11, 9, kValue.i32.negative.min, kValue.i32.positive.max - 1, -1]),
+      `
+    var a: i32 = -10;
+    var b: i32 = 10;
+    var c: i32 = ${kValue.i32.negative.min + 1};
+    var d: i32 = ${kValue.i32.positive.max};
+    var e: i32 = 0;
+
+    a--;
+    b--;
+    c--;
+    d--;
+    e--;
+
+    push_output(a);
+    push_output(b);
+    push_output(c);
+    push_output(d);
+    push_output(e);
+`
+    );
+  });
+
+g.test('scalar_i32_decrement_underflow')
+  .desc('Tests decrement of scalar i32 values which underflow')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'i32',
+      new Int32Array([kValue.i32.positive.max]),
+      `
+    var a: i32 = ${kValue.i32.negative.min};
+    a--;
+    push_output(a);
+`
+    );
+  });
+
+g.test('scalar_u32_decrement')
+  .desc('Tests decrement of scalar u32 values')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'u32',
+      new Uint32Array([0, 9, kValue.u32.max - 1]),
+      `
+    var a: u32 = 1;
+    var b: u32 = 10;
+    var c: u32 = ${kValue.u32.max};
+
+    a--;
+    b--;
+    c--;
+
+    push_output(a);
+    push_output(b);
+    push_output(c);
+`
+    );
+  });
+
+g.test('scalar_u32_decrement_underflow')
+  .desc('Tests decrement of scalar u32 values which underflow')
+  .fn(t => {
+    runStatementTest(
+      t,
+      'u32',
+      new Uint32Array([kValue.u32.max]),
+      `
+    var a: u32 = 0;
+    a--;
+    push_output(a);
+`
+    );
+  });
+
+g.test('vector_element_increment')
```

Comment:
Note for myself: ++ and -- are only for integer scalars, so there is no (vec)++ meaning ++ is applied component-wise sugar